lib/server.ml

```
@@ -266,6 +278,22 @@ let rec input_userauth_request t username service auth_method =
     else
       handle_auth t
 
+let reject_userauth t _userauth =
+  match t.auth_state with
+  | Auth.Inprogress (u, s, nfailed) ->
+    let t = { t with auth_state = Auth.Inprogress (u, s, succ nfailed) } in
+    Ok (t, Ssh.Msg_userauth_failure ([ "publickey"; "password" ], false))
+  | Auth.Done | Auth.Preauth ->
+    Error "userauth in unexpected state"
```

I don't know if we should raise invalid argument here instead?!